topic-operator/src/main/java/io/strimzi/operator/topic/K8sTopicWatcher.java

```diff
@@ -31,7 +31,7 @@ public K8sTopicWatcher(TopicOperator topicOperator, LabelPredicate resourcePredi
     public void eventReceived(Action action, KafkaTopic kafkaTopic) {
         ObjectMeta metadata = kafkaTopic.getMetadata();
         Map<String, String> labels = metadata.getLabels();
-        if (resourcePredicate.test(kafkaTopic)) {
+        if (resourcePredicate.test(kafkaTopic) || kafkaTopic.getMetadata().getLabels() == null || kafkaTopic.getMetadata().getLabels().size() == 0) {
```

Comment:
I think that the idea of this issue is to let the user (or whoever deploys the TO) to:
* Either select the labels which should be set on the resources (such as `strimzi.io/cluster. my-cluster`)
* Or set that no labels should be required in which case it should not check any labels.

You seem to use `||` everywhere which sounds like a user who specifies that he wants it t process KafkaTopics with labels `strimzi.io/cluster. my-cluster` would suddenly start consuming also those without any labels. That was not the intention.

Also, the idea with the label selector being not configured is not that there could be no labels on the KafkaTopics, but that we don't care about them.